src/numbers/fixed_point/fp16/core.cairo

```diff
@@ -0,0 +1,167 @@
+use core::debug::PrintTrait;
+
+use orion::numbers::fixed_point::core::FixedTrait;
+use orion::numbers::fixed_point::implementations::fp16x16::math::{
+    core as core_math, trig, hyp, erf
+};
+use orion::numbers::fixed_point::utils;
+
+
+// CONSTANTS
+const TWO: u32 = 131072; // 2 ** 17
+const ONE: u32 = 65536; // 2 ** 16
+const HALF: u32 = 32768; // 2 ** 15
+const MAX: u32 = 2147483648; // 2 ** 31
+
+impl fp16Impl of FixedTrait<i32> {
+    fn ZERO() -> i32 {
+       0
+    }
+
+    fn HALF() -> i32 {
+       HALF
```

Comment:
`HALF` constant is type `u32` but the returned type is `i32`. you should change constant type